BEP100.md

```diff
@@ -0,0 +1,137 @@
++ Web3 API - for basic compatibility with Web3 ecosystem including MetaMask and other applications;
++ Transaction Pool - for managing internal BAS policies for transaction filtering and for charging fees for the system operational;
++ PoA & PoS Consensus - for users to be able to vote for the honest validators in the BAS network and guarantee the safeness of actions applied on the chain;
++ Storage & State - for persisting local data.
+
+### 5.0 Circulation Model and Native Asset Bridge
+
+The key parts of each BAS application are native token circulation model and cross chain bridge for native assets. Native assets of the BAS chain are located in the BAS application and managed by sidechain directly. BAS is designed to provide cross chain functionality for the native assets. Since native assets are fully managed by BAS developers they can compromise token supply or mint/burn tokens. We leave such a decision on the conscience of the developers, because by manipulating the supply of their tokens they only risk their reputation. The only thing required for native cross chain bridge is block header verification that allows one to keep an active validator set and verify the correctness of cross chain transactions.
+
+BAS is not designed to have a BSC-compatible consensus or EVM execution environment, because BAS is technology-agnostic solution. Here we provide only basic functionality for the BAS applications and help them to set up their work and be a part of the BSC ecosystem. We can’t verify each operation on the chain and here we fully trust BAS developers, but we must be sure that validator transition is not compromised that is strictly required for the cross chain operations for the native asset. To reach this we’d like to introduce a block header verification function (BHVF) that must be specified by the BAS development team.
+
+BHVF is a function in the Solidity language that is able to verify block headers from BAS applications. BHVF can verify only validator transition and can be used to verify the correctness of the data that is provided in the BAS, but it can't prove the correctness of the state transition. That is why validator is set is required. BHVF is one of the required parameters to register BAS in the BSC’s smart contract to bring native cross chain functionality.
+
+Block header verification is not a very complicated function, but we will have to verify block headers to be sure that validator transition is well-done. Also by using BLS/BN we don’t have to pass all block headers into it to reach confirmation, but enough to publish only epoch blocks that contain new validator sets and signatures from all previous validators.
+
+```solidity
+interface IProofVerificationFunction {
+
+    function verifyValidatorTransition(bytes[] calldata proofs, uint256 chainId, address[] calldata existingValidatorSet) external view returns (address[] memory newValidatorSet);
```

Comment:
What are the `proofs` here means? Are they the N blocks after the epoch?

Reply:
I guess `existingValidatorSet` should store on chain, otherwise how can we trust `existingValidatorSet` if it is an input parameter?

Reply:
You have described how to update the validator set here,  should we have a chapter to describe how to bridge common cross-chain packages, a common interface is needed as well?